spec/services/authentication_spec.rb

```diff
@@ -0,0 +1,99 @@
+describe Authentication do
+  describe '.call' do
+    let(:user) { create(:user) }
+
+    subject { described_class.call(encode_token(payload)) }
+
+    def encode_token(payload)
+      JWT.encode(
+        payload,
+        Settings.authentication.secret,
+        Settings.authentication.algorithm
+      )
+    end
+
+    context 'with a DfE-SignIn ID and email' do
+      let(:payload) do
+        {
+          email:           user.email,
+          sign_in_user_id: user.sign_in_user_id
+        }
+      end
+
+      it { should eq user }
+    end
+
+    context 'with a DfE-SignIn ID but no email' do
+      let(:payload) { { sign_in_user_id: user.sign_in_user_id } }
+
+      it { should eq user }
+    end
+
+    context 'with an email but no DfE-SignIn ID' do
+      let(:payload) { { email: user.email } }
+
+      it { should eq user }
```

Comment:
I don't think this should ever be allowed, as the user must signed in via dfe sign in to progress

The original use case for the user email address was used as a `whitelist` but they must of signed in via dfe sign
If the user email is a mismatch from db but has dfe sign in id let them though and update their original email address on end.

Reply:
Gotcha, the correct case here is if the user has a DfE-SignIn ID but it doesn't match our records, we fall back to the email. But if there's no DfE-SignIn ID we shouldn't let them in.

When we fall back to the email, we should update the DfE-SignIn ID in the database however. I'll make these changes now.